test/backtrace.jl

```diff
@@ -258,3 +258,76 @@ let code = """
     @test occursin("InterpreterIP in top-level CodeInfo for Main.A", bt_str)
 end
 
+"""
+    _reformat_sp(bt_data...) -> sp::Vector{Ptr{Cvoid}}
+
+Convert the output `bt_data` of `jl_backtrace_from_here` with `returnsp` flag set to a
+vector of valid stack pointers `sp`; i.e., `sp` is a subset of `bt_data[3]`.
+
+See also `Base._reformat_bt`.
+"""
+function _reformat_sp(
+    bt_raw::Array{Ptr{Cvoid},1},
+    bt2::Array{Any,1},
+    sp_raw::Array{Ptr{Cvoid},1},
+)
+    bt = Base._reformat_bt(bt_raw, bt2)
+    sp = empty!(similar(sp_raw))
+    i = j = 0
+    while true
+        # Advance `i` such that `bt[i] isa Ptr{Cvoid}` (native pointer).
+        local ip
+        while true
+            if i == lastindex(bt)
+                return sp
+            end
+            i += 1
+            x = bt[i]
+            if x isa Ptr{Cvoid}
+                ip = x
+                break
+            end
+        end
+        # Advance `j` such that `bt_raw[j] == bt[i]` to find a valid stack pointer.
+        while true
+            if j == lastindex(bt_raw)
+                return sp
+            end
+            j += 1
+            if bt_raw[j] == ip
+                push!(sp, sp_raw[j])
+                break
+            end
+        end
+    end
+end
+
+"""
+    withframeaddress(f)
+
+Call function `f` with an address `ptr::Ptr{Cvoid}` of an independent frame
+immediately outer to `f`.
+"""
+@noinline function withframeaddress(f)
+    sp = ccall("llvm.frameaddress", llvmcall, Ptr{Cvoid}, (Int32,), 0)
```

Comment:
the return type here is wrong (it should be `i8*`)

Reply:
Probably need to do this instead:
```julia
julia> @eval frameaddr() = Core.Intrinsics.llvmcall(($"""
       declare i8* @llvm.frameaddress(i32)
       define i$(Sys.WORD_SIZE) @frameaddr() {
         %1 = call i8* @llvm.frameaddress(i32 0)
         %2 = ptrtoint i8* %1 to i$(Sys.WORD_SIZE)
         ret i$(Sys.WORD_SIZE) %2
       }""", "frameaddr"), UInt, Tuple{})
frameaddr (generic function with 1 method)

julia> frameaddr()
0x00007ffdac2bd2d0    
```

Reply:
added 715c673a889a2c8a9b81b5628db9e383ab47b8c9